### PR TITLE
Documentation: Add block submission guidelines.

### PR DESCRIPTION
**Pull request by StevenDufresne**
## Description
This PR exposes documentation to help block developers submit blocks to the block directory in the hope of getting more collaboration. The end document doesn't need to live where this PR proposes it.  

This PR is associated to #23484

### CODE REVIEW

File: docs/designers-developers/developers/tutorials/create-block/submitting-to-block-directory.md [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+- carousel
+- gallery
+
+[Read more](https://github.com/WordPress/gutenberg/blob/master/docs/rfc/block-registration.md#keyword) about keywords.
+
+### Choose the right category
+Gutenberg allows you to indicate which category your block fits in. Choosing the right category makes it easy for users to find in the Gutenberg menu.
+
+**Possible Values**:
+- Common
```
Comment (gziolo) [inline docs/designers-developers/developers/tutorials/create-block/submitting-to-block-directory.md:52]: This list is outdated, see: https://github.com/WordPress/gutenberg/pull/19279

Comment (StevenDufresne) [reply docs/designers-developers/developers/tutorials/create-block/submitting-to-block-directory.md:52]: This list was taken from the [block-registration documentation](https://github.com/WordPress/gutenberg/blob/master/docs/rfc/block-registration.md#category). I'll update it there as well.

File: docs/designers-developers/developers/tutorials/create-block/submitting-to-block-directory.md [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+
+**Guidelines**:
+
+- Name your block based on what it does
+- Clearly describe your block
+- Add Keywords for all contexts
+- Choose the right category
+
+### Name your block based on what it does
+Users typically search the block directory within gutenberg and do so in the context of a task. For example, when building their post, a user may search the Block Directory for an “image gallery”. Naming your block accordingly will help the Block Directory surface it when it's needed.
```
Comment (itsjusteileen) [inline docs/designers-developers/developers/tutorials/create-block/submitting-to-block-directory.md:21]: Remove gutenberg in favor of block editor or editor window
"Users typically search the block directory directly from the editor, doing so when actively editing a post. For example, a user may search...

File: docs/designers-developers/developers/tutorials/create-block/submitting-to-block-directory.md [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+## Step 2: Analyze your plugin
+
+The `block.json` file lives in the root folder of your block and provides the Block Directory and Gutenberg important information about your block. Along with being the place to store contextual information about your block like the: `name`, `description`, `keywords` and `category`, the `block.json` file stores the location of your block’s files.
+
+Double check that the following is true for your block:
+`editorScript` is pointing to the JavaScript bundle that includes all the code used in the **editor**.
+`editorStyle` is pointing to the CSS bundle that includes all the css used in the **editor**.
+`script` is pointing to the JavaScript bundle that includes all the code used on the **website**.
+`style` is pointing to the CSS bundle that includes all the code used on the **website**.
+I have included example data (Optional, but useful)
```
Comment (itsjusteileen) [inline docs/designers-developers/developers/tutorials/create-block/submitting-to-block-directory.md:72]: Include example data (optional, but useful).

File: docs/designers-developers/developers/tutorials/create-block/submitting-to-block-directory.md [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+
+Examples for an Image Slider block:
+- slider
+- carousel
+- gallery
+
+[Read more](https://github.com/WordPress/gutenberg/blob/master/docs/rfc/block-registration.md#keyword) about keywords.
+
+### Choose the right category
+Gutenberg allows you to indicate which category your block fits in. Choosing the right category makes it easy for users to find in the Gutenberg menu.
```
Comment (itsjusteileen) [inline docs/designers-developers/developers/tutorials/create-block/submitting-to-block-directory.md:49]: The Block Editor allows you to indicate the category your block belongs in, making it easier for users to locate your block in the menu.

**Comment by itsjusteileen** [comment]
Can programmatic block templates aka '[inner blocks](https://developer.wordpress.org/block-editor/developers/block-api/block-templates/)' also be included in the Block Directory? If so mention here would be noteworthy.

File: docs/designers-developers/developers/tutorials/create-block/submitting-to-block-directory.md [hunk 5]
```diff
@@ -0,0 +1,88 @@
+#Share your Gutenberg Block with the World
+So you’ve created an awesome block have you? Care to share? Here is some basic information on how to submit a block to the Block Directory.
```
Comment (annezazu) [inline docs/designers-developers/developers/tutorials/create-block/submitting-to-block-directory.md:2]: I would just say simply: "So you've created an awesome block? Care to share?"

File: docs/designers-developers/developers/tutorials/create-block/submitting-to-block-directory.md [hunk 6]
```diff
@@ -0,0 +1,88 @@
...
+
+Need help building a block? Check out our block tutorials to get started.
```
Comment (annezazu) [inline docs/designers-developers/developers/tutorials/create-block/submitting-to-block-directory.md]: I would directly link to the documentation for block creation: https://github.com/WordPress/gutenberg/tree/master/docs/designers-developers/developers/tutorials/create-block

**Comment by annezazu** [comment]
Just looked this over! I'm curious where people could go to get help if they get stuck in the submission process. If that's something that can be defined right now,  I think it would be super helpful to include. In theory, doing so, should help get  more feedback to you all about what blockers there are for submitting to the directory and increase the chances someone will complete a block directory submission 💥

File: docs/designers-developers/developers/tutorials/create-block/submitting-to-block-directory.md (hunk 3 above)
Comment (StevenDufresne) [reply docs/designers-developers/developers/tutorials/create-block/submitting-to-block-directory.md:72]: I changed it to `Includes example data (optional, but useful).` because it makes more sense in the context of the `Double check that the following is true for your block`. It feels more natural to me. I could be wrong though? :)

**Comment by StevenDufresne** [comment]
> Can programmatic block templates aka '[inner blocks](https://developer.wordpress.org/block-editor/developers/block-api/block-templates/)' also be included in the Block Directory? If so mention here would be noteworthy.

**Comment by StevenDufresne** [comment]
> Can programmatic block templates aka '[inner blocks](https://developer.wordpress.org/block-editor/developers/block-api/block-templates/)' also be included in the Block Directory? If so mention here would be noteworthy.

This seems like it would be a technical details of creating a block. This document wasn't designed to be take on the technical details of a block directory block but I can see how that needs to live somewhere and right now it doesn't.

File: docs/designers-developers/developers/tutorials/create-block/submitting-to-block-directory.md (hunk 6 above)
Comment (StevenDufresne) [reply docs/designers-developers/developers/tutorials/create-block/submitting-to-block-directory.md]: I actually removed this line with the expectation that this document will tie closely into the `Finishing Touches` document and therefore may not be necessary. https://github.com/WordPress/gutenberg/issues/23484.

I'll add it back in later if we think it necessary.

**Comment by StevenDufresne** [comment]
@annezazu @itsjusteileen Very thankful for the reviews! :)

File: docs/designers-developers/developers/tutorials/create-block/submitting-to-block-directory.md [hunk 7]
```diff
@@ -0,0 +1,86 @@
+#Share your Block with the World
```
Comment (mkaz) [inline docs/designers-developers/developers/tutorials/create-block/submitting-to-block-directory.md:1]: I've got some minor formatting updates

```suggestion
# Share your Block with the World
```

File: docs/designers-developers/developers/tutorials/create-block/submitting-to-block-directory.md [hunk 8]
```diff
@@ -0,0 +1,86 @@
...
+So you've created an awesome block? Care to share?
+
+**Contents**:
+1. Help users understand your block
+2. Analyze your plugin
+3. Zip & Submit
+
+
```
Comment (mkaz) [inline docs/designers-developers/developers/tutorials/create-block/submitting-to-block-directory.md:9]: No double blank lines.

```suggestion
```

File: docs/designers-developers/developers/tutorials/create-block/submitting-to-block-directory.md [hunk 9]
```diff
@@ -0,0 +1,86 @@
...
+**Guidelines**:
+
+- Name your block based on what it does
+- Clearly describe your block
+- Add Keywords for all contexts
+- Choose the right category
+
+### Name your block based on what it does
+Users typically search the block directory within the Block Editor and do so in the context of a task. For example, when building their post, a user may search the Block Directory for an “image gallery”. Naming your block accordingly will help the Block Directory surface it when it's needed.
```
Comment (mkaz) [inline docs/designers-developers/developers/tutorials/create-block/submitting-to-block-directory.md:21]: Blank line between heading and paragraph

```suggestion

Users typically search the block directory within the Block Editor and do so in the context of a task. For example, when building their post, a user may search the Block Directory for an “image gallery”. Naming your block accordingly will help the Block Directory surface it when it's needed.
```

File: docs/designers-developers/developers/tutorials/create-block/submitting-to-block-directory.md [hunk 10]
```diff
@@ -0,0 +1,86 @@
...
+
+**Not So Good**: WebTeam5 Image Works
+**Good**: Responsive Image Slider by WebTeam5
+
+**Question: What happens when there are multiple blocks with similar names?**
+Try your best to make your block's name functional and unique to make it stand out. Look for applicable synonyms or include a prefix if necessary.
+
+
```
Comment (mkaz) [inline docs/designers-developers/developers/tutorials/create-block/submitting-to-block-directory.md:29]: No blank line

```suggestion
```

File: docs/designers-developers/developers/tutorials/create-block/submitting-to-block-directory.md [hunk 11]
```diff
@@ -0,0 +1,86 @@
...
+### Clearly describe your block
+The description really helps to communicate what your block does.The quicker a user understands how your block will help them, the more likely it is a user will use your block. Users will be reading your block's description within the Block Editor where space can be limited. Try to keep it short and concise.
```
Comment (mkaz) [inline docs/designers-developers/developers/tutorials/create-block/submitting-to-block-directory.md:31]: Blank line between heading and paragraph

```suggestion

The description really helps to communicate what your block does.The quicker a user understands how your block will help them, the more likely it is a user will use your block. Users will be reading your block's description within the Block Editor where space can be limited. Try to keep it short and concise.
```

File: docs/designers-developers/developers/tutorials/create-block/submitting-to-block-directory.md [hunk 12]
```diff
@@ -0,0 +1,86 @@
...
+
+Examples for an Image Slider block:
+- slider
+- carousel
+- gallery
+
+[Read more](https://github.com/WordPress/gutenberg/blob/master/docs/rfc/block-registration.md#keyword) about keywords.
+
+### Choose the right category
+The Block Editor allows you to indicate the category your block belongs in, making it easier for users to locate your block in the menu.
```
Comment (mkaz) [inline docs/designers-developers/developers/tutorials/create-block/submitting-to-block-directory.md:49]: Blank line between heading and paragraph

```suggestion

The Block Editor allows you to indicate the category your block belongs in, making it easier for users to locate your block in the menu.
```

File: docs/designers-developers/developers/tutorials/create-block/submitting-to-block-directory.md [hunk 13]
```diff
@@ -0,0 +1,86 @@
...
+- media
+- design
+- widgets
+- embed
+
+[Read more](https://github.com/WordPress/gutenberg/blob/master/docs/rfc/block-registration.md#category) about categories.
+
+Wondering where to input all this information? Read the next section :)
+
+
```
Comment (mkaz) [inline docs/designers-developers/developers/tutorials/create-block/submitting-to-block-directory.md:62]: No double blank lines.

```suggestion
```

File: docs/designers-developers/developers/tutorials/create-block/submitting-to-block-directory.md [hunk 14]
```diff
@@ -0,0 +1,86 @@
...
+`style` is pointing to the CSS bundle that includes all the code used on the **website**.
+Includes example data (Optional, but useful)
+
+Although it isn’t necessary that you have both an `editorScript/editorStyle` and `script/style` listed in your `block.json` we encourage the separation of code used for editing and code used for displaying your block in order to keep both interfaces running quickly.
+
+The `block.json` file also contains other important properties. Take a look at an [example block.json](https://github.com/WordPress/gutenberg/blob/master/docs/rfc/block-registration.md#registering-a-block-type) file for reference.
+
+
+## Step 3: Zip & Submit
+The community is ecstatic you made it this far! Time to submit your plugin!
```
Comment (mkaz) [inline docs/designers-developers/developers/tutorials/create-block/submitting-to-block-directory.md:80]: Blank line between heading and paragraph

```suggestion

The community is ecstatic you made it this far! Time to submit your plugin!
```

**Review by mkaz — APPROVED** [review]
I have a few minor formatting tweaks, not a big deal.

This is good, I'm good with adding and iterating, this tutorial is not linked up yet, so it is a good to commit and add to.

👍

File: docs/designers-developers/developers/tutorials/create-block/submitting-to-block-directory.md [hunk 15]
```diff
@@ -0,0 +1,89 @@
...
+Providing straightforward, easy to understand block information is important to the block directory and our end users.
+
+**Guidelines**:
+
+- Name your block based on what it does
+- Clearly describe your block
+- Add Keywords for all contexts
+- Choose the right category
+
+### Name your block based on what it does
```
Comment (tellyworth) [inline docs/designers-developers/developers/tutorials/create-block/submitting-to-block-directory.md:21]: I think we should take a clearer position on block names (specifically names not titles). Especially the namespace part. Unsurprisingly there's not much consistency in the plugin directory.

At a quick glance, I see people using namespaces that are:
* The plugin slug (like `card-block/main`, `pdf-viewer-block/standard`)
* The block name (`blockasaurus/blockasaurus`)
* The developer's employer (`a8c/waves`)
* The author's nickname or initials (`cgb/block-importdoc-block`)

None of these are wrong because we haven't given anyone clear guidance as to what they should use. I don't have a strong opinion, but I do thing the first two dot points above are sub-optimal. Using a very generic suffix like `blah/main` or `blah/standard` seems unhelpful. So does repetition like `blah/blah`.

Maybe we should recommend developers pick a namespace to use across all of their blocks, like in the last two dot points?

File: docs/designers-developers/developers/tutorials/create-block/submitting-to-block-directory.md [hunk 16]
```diff
@@ -0,0 +1,89 @@
...
+The `block.json` file lives in the root folder of your block and provides the Block Directory important information about your block. Along with being the place to store contextual information about your block like the: `name`, `description`, `keywords` and `category`, the `block.json` file stores the location of your block’s files.
+
+Double check that the following is true for your block:
+`editorScript` is pointing to the JavaScript bundle that includes all the code used in the **editor**.
+`editorStyle` is pointing to the CSS bundle that includes all the css used in the **editor**.
+`script` is pointing to the JavaScript bundle that includes all the code used on the **website**.
+`style` is pointing to the CSS bundle that includes all the code used on the **website**.
+Includes example data (Optional, but useful)
+
+Although it isn’t necessary that you have both an `editorScript/editorStyle` and `script/style` listed in your `block.json` we encourage the separation of code used for editing and code used for displaying your block in order to keep both interfaces running quickly.
```
Comment (tellyworth) [inline docs/designers-developers/developers/tutorials/create-block/submitting-to-block-directory.md:77]: I think you could be clearer in stating that at least one .js file is definitely required.

**Comment by bfintal** [comment]
This should probably be adjusted as well since some topics overlap https://github.com/WordPress/wporg-plugin-guidelines/blob/block-guidelines/blocks.md

**Comment by StevenDufresne** [comment]
> This should probably be adjusted as well since some topics overlap https://github.com/WordPress/wporg-plugin-guidelines/blob/block-guidelines/blocks.md

Hey @bfintal! Can you provide more information so we can focus on specific changes?

This document is not considered a standalone document and will contain topics that are mentioned both in block creation topics and block submission/plugin guidelines documentation.

File: docs/designers-developers/developers/tutorials/create-block/submitting-to-block-directory.md (hunk 15 above)
Comment (StevenDufresne) [reply docs/designers-developers/developers/tutorials/create-block/submitting-to-block-directory.md:21]: I am tempted to suggest the last 2 are suboptimal because company names and initials are more likely to be repeated where as the block's folder name must be unique. Are there any issues if we suggest the namespace must match the plugin directory folder?

File: docs/designers-developers/developers/tutorials/create-block/submitting-to-block-directory.md (hunk 16 above)
Comment (StevenDufresne) [reply docs/designers-developers/developers/tutorials/create-block/submitting-to-block-directory.md:77]: What about something like:

"Although blocks require at least one javascript file listed in the `block.json`, we encourage the separation of code used for editing and code used for displaying your block in order to keep both interfaces running quickly.

File: docs/designers-developers/developers/tutorials/create-block/submitting-to-block-directory.md (hunk 15 above)
Comment (ryelle) [reply docs/designers-developers/developers/tutorials/create-block/submitting-to-block-directory.md:21]: I definitely think we should avoid `cgb` or other starter-content prefixes, since those are much more likely to collide.

The plugin slug will be unique in order to be uploaded to the directory, so that makes sense as a namespace to me. But I also think company name or username are fine too, and it can be left to the company/person to make sure they're not uploading multiple blocks with the same name. Initials are probably more likely to collide between people though, so I'd say discourage that.

Comment (StevenDufresne) [reply docs/designers-developers/developers/tutorials/create-block/submitting-to-block-directory.md:21]: Another we should consider is that there is a desire to surface the block's name:
https://github.com/WordPress/gutenberg/issues/16521

Comment (StevenDufresne) [reply docs/designers-developers/developers/tutorials/create-block/submitting-to-block-directory.md:21]: Surfacing the block's name, to me, suggests that the unique plugin slug is the best route.

File: docs/designers-developers/developers/tutorials/create-block/submitting-to-block-directory.md (hunk 3 above)
Comment (ryelle) [reply docs/designers-developers/developers/tutorials/create-block/submitting-to-block-directory.md]: I had to read this multiple times before I realized this meant the `example` property in block.json. I thought it was a note to yourself to come back and add example data 😅 

Is the example data a requirement (or do we think it will be a requirement in the future)? We should either explain this better or drop it, IMO

File: docs/designers-developers/developers/tutorials/create-block/submitting-to-block-directory.md [hunk 17]
```diff
@@ -0,0 +1,102 @@
+# Share your Block with the World
+
+So you've created an awesome block? Care to share?
+
+**Contents**:
+1. Help users understand your block
+2. Analyze your plugin
```
Comment (ryelle) [inline docs/designers-developers/developers/tutorials/create-block/submitting-to-block-directory.md:7]: What do you think about renaming this section to "Provide data to the block directory"? Something about "Analyze your plugin" makes it sound automated, rather than curating data and describing your block(s)

**Comment by ryelle** [comment]
I added more details about the block.json, including an example of a very simple block.json file in https://github.com/WordPress/gutenberg/pull/23545/commits/cb7e2e6f82e3f499fad1c3c70513897afa98e45c

File: docs/designers-developers/developers/tutorials/create-block/submitting-to-block-directory.md (hunk 15 above)
Comment (ryelle) [reply docs/designers-developers/developers/tutorials/create-block/submitting-to-block-directory.md:21]: So is this decided, we'll _require_ the plugin slug to be used as a prefix? Should that be added into the guidelines, or is it enough to add a check to the Block Checker?

File: docs/designers-developers/developers/tutorials/create-block/submitting-to-block-directory.md (hunk 17 above)
Comment (paaljoachim) [reply docs/designers-developers/developers/tutorials/create-block/submitting-to-block-directory.md:7]: Here are some of my suggestions.

# Share your Block

Share your block with others.

##Contents**:
1. Help users understand your block
2. Analyze your plugin? --> What does that mean? Using basic words to explain what it is.
Does this mean explain what your plugin does? Explain how you made your plugin? 

@ryelle Kelly
Curating data I really do not understand what that means.

File: docs/designers-developers/developers/tutorials/create-block/submitting-to-block-directory.md [hunk 18]
```diff
@@ -0,0 +1,102 @@
...
+3. Zip & Submit
+
+## Step 1: Help users understand your block
+
+Providing straightforward, easy to understand block information is important to the block directory and our end users.
+
```
Comment (paaljoachim) [inline docs/designers-developers/developers/tutorials/create-block/submitting-to-block-directory.md:13]: It is important to the Block Directory and our end users to provide easy to understand information on how your block was created.

File: docs/designers-developers/developers/tutorials/create-block/submitting-to-block-directory.md (hunk 15 above)
Comment (paaljoachim) [reply docs/designers-developers/developers/tutorials/create-block/submitting-to-block-directory.md:21]: "Name your block based on what it does"

It would be good with some example use cases to give a more detailed approach to how someone should approach the naming of their block. Have more detailed guidelines to a naming scheme as you are talking about above here. 

The guidelines also need to show examples of naming, describing, keywords and choosing the right category.
That can mean use a Create a Block documentation they should follow in getting these things right. (At present I do not know if the Create a Block documentation gives this needed information.)

Bottom line is that we can not assume a person will understand but we need to give clear guidelines to how things should be done.

File: docs/designers-developers/developers/tutorials/create-block/submitting-to-block-directory.md (hunk 9 above)
Comment (paaljoachim) [reply docs/designers-developers/developers/tutorials/create-block/submitting-to-block-directory.md:23]: Another suggestion.

Users might search the Block Directory (capital B? and capital D?) within the Block Editor (Block Editor uses capital letters and I wonder if the Block Directory should do the same).

Users might search the Block Directory within the Block Editor to find a block they might need.

File: docs/designers-developers/developers/tutorials/create-block/submitting-to-block-directory.md [hunk 19]
```diff
@@ -0,0 +1,102 @@
...
+- Add Keywords for all contexts
+- Choose the right category
+
+### Name your block based on what it does
+
+Users typically search the block directory within the Block Editor and do so in the context of a task. For example, when building their post, a user may search the Block Directory for an “image gallery”. Naming your block accordingly will help the Block Directory surface it when it's needed.
+
+**Not So Good**: WebTeam5 Image Works
+**Good**: Responsive Image Slider by WebTeam5
+
```
Comment (paaljoachim) [inline docs/designers-developers/developers/tutorials/create-block/submitting-to-block-directory.md:27]: It is very helpful to have additional examples.

File: docs/designers-developers/developers/tutorials/create-block/submitting-to-block-directory.md [hunk 20]
```diff
@@ -0,0 +1,102 @@
...
+**Question: What happens when there are multiple blocks with similar names?**
+Try your best to make your block's name functional and unique to make it stand out. Look for applicable synonyms or include a prefix if necessary.
+
```
Comment (paaljoachim) [inline docs/designers-developers/developers/tutorials/create-block/submitting-to-block-directory.md:30]: Here it would be good to just mention that it is good to include a prefix. (Regarding the conversation above this might be changed to remember to include a prefix.)

File: docs/designers-developers/developers/tutorials/create-block/submitting-to-block-directory.md (hunk 11 above)
Comment (paaljoachim) [reply docs/designers-developers/developers/tutorials/create-block/submitting-to-block-directory.md:33]: A description really helps communicate to the user what your block does. It helps them decide if the block is the correct one for them to use for the task they have in mind. Remember to keep the description short and concise as the space within the Block Editor is limited.

File: docs/designers-developers/developers/tutorials/create-block/submitting-to-block-directory.md [hunk 21]
```diff
@@ -0,0 +1,102 @@
...
+### Clearly describe your block
+
+The description really helps to communicate what your block does.The quicker a user understands how your block will help them, the more likely it is a user will use your block. Users will be reading your block's description within the Block Editor where space can be limited. Try to keep it short and concise.
+
+**Not So Good**: The best way to show images on your website using jQuery and CSS.
+**Good**: An easy to use, responsive, Image gallery block. 
+
```
Comment (paaljoachim) [inline docs/designers-developers/developers/tutorials/create-block/submitting-to-block-directory.md:37]: A responsive image gallery block. (easy to use well that is up to the user to decide.)

File: docs/designers-developers/developers/tutorials/create-block/submitting-to-block-directory.md [hunk 22]
```diff
@@ -0,0 +1,102 @@
...
+**Tip**: It’s not about marketing your block, in fact we want to avoid marketing in blocks. You can read more about it in the [plugin guidelines]. Stick to being as clear as you can. The block directory will provide metrics to let users know how awesome your block is!
+
```
Comment (paaljoachim) [inline docs/designers-developers/developers/tutorials/create-block/submitting-to-block-directory.md:39]: The above is not clear for me. 
I would skip the first sentence, and instead add something like the below.

You can read more about marketing your block in the plugin guidelines. The Block Directory will provide metrics to let users know helpful your block is.

**Comment by paaljoachim** [comment]
When creating any tutorial it is good to keep the language neutral and simple to understand.
As we are guiding someone through to understand how to do something. Having a neutral to the point voice in doing so is very important.

File: docs/designers-developers/developers/tutorials/create-block/submitting-to-block-directory.md [hunk 23]
```diff
@@ -0,0 +1,102 @@
...
+Keywords add extra context to your block and make it more likely to be found in the inserter. 
+
+Examples for an Image Slider block:
+- slider
+- carousel
+- gallery
+
+[Read more about keywords.](https://github.com/WordPress/gutenberg/blob/master/docs/rfc/block-registration.md#keywords)
+
```
Comment (paaljoachim) [inline docs/designers-developers/developers/tutorials/create-block/submitting-to-block-directory.md:50]: This is neutral and to the point! Exactly what is needed. Have that in the back of the mind through all of the tutorial.

File: docs/designers-developers/developers/tutorials/create-block/submitting-to-block-directory.md (hunk 12 above)
Comment (paaljoachim) [reply docs/designers-developers/developers/tutorials/create-block/submitting-to-block-directory.md:53]: The Block Editor allows you to select the category your block will be included into. Making it easier for users to locate your block in the menu.

**Comment by paaljoachim** [comment]
Do remember to keep the language as simple as possible.
Keep it also detailed with description.

Block Directory = Capital B and capital D. General adjustment through the documentation.

File: docs/designers-developers/developers/tutorials/create-block/submitting-to-block-directory.md [hunk 24]
```diff
@@ -0,0 +1,102 @@
...
+- widgets
+- embed
+
+[Read more about categories.](https://github.com/WordPress/gutenberg/blob/master/docs/rfc/block-registration.md#category)
+
+Wondering where to input all this information? Read the next section :)
+
+## Step 2: Analyze your plugin
+Each block in your plugin should have a corresponding `block.json` file. This file provides the Block Directory important information about your block. Along with being the place to store contextual information about your block like the: `name`, `description`, `keywords` and `category`, the `block.json` file stores the location of your block’s files.
+
```
Comment (paaljoachim) [inline docs/designers-developers/developers/tutorials/create-block/submitting-to-block-directory.md:68]: Each block in your plugin needs to have a corresponding `block.json` file. This file provides important information about your block to the Block Directory.  

What important information? Go straight to the point and give the details without it saying the word important. The files stores such and such information. Give a detailed overview of what the json file stores.

Kinda like.
The `block.json` file stores the location of your block's files as well as contextual information like the: `name`, `description`, `keywords` and `category`. In addition is also stores.....?

File: docs/designers-developers/developers/tutorials/create-block/submitting-to-block-directory.md [hunk 25]
```diff
@@ -0,0 +1,102 @@
...
+Block plugins submitted to the Block Directory can contain mutliple blocks only if they are children of a single parent/ancestor. There should only be one main block. For example, a list block can contain list-item blocks. Children blocks must set the `parent` property in their `block.json` file.
+
```
Comment (paaljoachim) [inline docs/designers-developers/developers/tutorials/create-block/submitting-to-block-directory.md:70]: Block plugins submitted to the Block Directory can contain **multiple** (typo) blocks only if they are children of a single parent/ancestor. 

Rephrasing.
Block plugins submitted to the Block Directory should have one main block, but it can contain multiple child blocks. 
For example, a list block can contain list-item child blocks. Child blocks must set the `parent` property in their `block.json` file.

File: docs/designers-developers/developers/tutorials/create-block/submitting-to-block-directory.md [hunk 26]
```diff
@@ -0,0 +1,102 @@
...
+Double check that the following is true for your block:
+
```
Comment (paaljoachim) [inline docs/designers-developers/developers/tutorials/create-block/submitting-to-block-directory.md:72]: Double check the following:

File: docs/designers-developers/developers/tutorials/create-block/submitting-to-block-directory.md (hunk 16 above)
Comment (paaljoachim) [reply docs/designers-developers/developers/tutorials/create-block/submitting-to-block-directory.md:78]: We encourage the separation of code used in the backend and the frontend to keep both interfaces running smoothly by including both the following files: `editorScript/editorStyle` and `script/style` listed in your `block.json`

File: docs/designers-developers/developers/tutorials/create-block/submitting-to-block-directory.md [hunk 27]
```diff
@@ -0,0 +1,102 @@
...
+- `editorScript` is pointing to the JavaScript bundle that includes all the code used in the **editor**.
+- `editorStyle` is pointing to the CSS bundle that includes all the css used in the **editor**.
+- `script` is pointing to the JavaScript bundle that includes all the code used on the **website**.
+- `style` is pointing to the CSS bundle that includes all the code used on the **website**.
+
+Although it isn’t necessary that you have both an `editorScript/editorStyle` and `script/style` listed in your `block.json` we encourage the separation of code used for editing and code used for displaying your block in order to keep both interfaces running quickly.
+
+The most basic `block.json` you need looks like this.
+
```
Comment (paaljoachim) [inline docs/designers-developers/developers/tutorials/create-block/submitting-to-block-directory.md:81]: Here is an example of a basic `block.json` file.

File: docs/designers-developers/developers/tutorials/create-block/submitting-to-block-directory.md [hunk 28]
```diff
@@ -0,0 +1,102 @@
...
+  "title": "Responsive Image Slider",
+  "description": "A responsive and easy to use image gallery block.",
+  "keywords": [ "slider", "carousel", "gallery" ],
+  "category": "media",
+  "editorScript": "file:./dist/editor.js",
+}
+```
+
+The `block.json` file also contains other important properties. Take a look at an [example block.json](https://github.com/WordPress/gutenberg/blob/master/docs/rfc/block-registration.md#registering-a-block-type) file for reference.
+
```
Comment (paaljoachim) [inline docs/designers-developers/developers/tutorials/create-block/submitting-to-block-directory.md:94]: Take a look at [example block.json] for additional properties to be included in the `block.json` file.

File: docs/designers-developers/developers/tutorials/create-block/submitting-to-block-directory.md [hunk 29]
```diff
@@ -0,0 +1,102 @@
...
+
+## Step 3: Zip & Submit
+
+The community is ecstatic you made it this far! Time to submit your plugin!
+
```
Comment (paaljoachim) [inline docs/designers-developers/developers/tutorials/create-block/submitting-to-block-directory.md:99]: The community is thankful for your contribution. Time to submit your plugin.

Let's remove the word ecstatic. It is a bit over the top.

File: docs/designers-developers/developers/tutorials/create-block/submitting-to-block-directory.md [hunk 30]
```diff
@@ -0,0 +1,102 @@
...
+Take a few moments to read [the block guidelines](https://github.com/WordPress/wporg-plugin-guidelines/blob/block-guidelines/blocks.md). Create a zip file of your block and navigate over to the [block plugin validator](https://wordpress.org/plugins/developers/block-plugin-validator/) and upload your plugin. 
+
+
```
Comment (paaljoachim) [inline docs/designers-developers/developers/tutorials/create-block/submitting-to-block-directory.md:102]: Go through [the block guidelines]. Create a zip file of your block and go to the [block plugin validator] and upload your plugin.

**Comment by paaljoachim** [comment]
Keep the language as simple and neutral as possible. Long sentences should likely be cut in two. Remember that people from all over the world with different English reading skills will be going through the documentation.

Block Directory = Capital B and capital D. General adjustment through the documentation.

**Comment by StevenDufresne** [comment]
> Keep the language as simple and neutral as possible. A neutral narrator voice. Long sentences should likely be cut in two. Remember that people from all over the world with different English reading skills will be going through the documentation.
> 
> Block Directory = Capital B and capital D. General adjustment through the documentation.
> 
> Ps When ready we need someone to go through the steps in a practical way. I will likely do so but I am following the steps of creating a local dev environment (#23593 still being worked on), then the creating your first block plugin (I do not know where the create a block docs is located right now - need to go through) and then finally submit your block (which I will get to).

Perfect. Thanks for the help!

File: docs/designers-developers/developers/tutorials/create-block/submitting-to-block-directory.md (hunk 22 above)
Comment (StevenDufresne) [reply docs/designers-developers/developers/tutorials/create-block/submitting-to-block-directory.md:39]: I think without that first sentence it would be hard to understand the context. I think It may be better to remove it instead.
